### ISSUE TITLE
Including dynamic post data with Dynamic Attributes

### ISSUE THREAD

**Issue by drosa88** [elementor/elementor#13341]
**What problem is your feature request going to solve? Please describe.**
I need a clean way to include dynamic post data into a grid of static elements. For example: I have 6 static cards in my grid. 2 of them should have the post_title, the featured_image and the_excerpt of the latest posts. Using the posts widget does not solve my problem because the design differences are too big.

**Describe the solution you'd like**
I would like another Dynamic Attribute like "POST QUERY". In the attribute settings I could choose a simple query like "Latest Post" and "Offset" for example. The result will be one single post. And for this post I can output the dynamic title, excerpt, featured image or something else.

What do you think? And is there a workaround until this feature maybe exists someday?

![Bildschirmfoto 2020-12-17 um 15 15 17](https://user-images.githubusercontent.com/6883050/102499042-ff661500-407a-11eb-8bd1-2b917f2ee3b5.jpg)

**Comment by Aviad-Herman** [comment]
Hey, great idea.
If I understand you request correctly, it's part of this thread #4440.
We do not have solution for this yet.

**Comment by drosa88** [comment]
Hey @Aviad-Herman, thanks for your message.

Mhh..not directly. There the focus is to add more functionality on the post widget to be able to create more individual designs.
But what I mean is to include some WP_Query logic into the Dynamic Attributes fields on **each** page.

To be clear I attach you a screenshot below:

![Bildschirmfoto 2020-12-17 um 15 15 17](https://user-images.githubusercontent.com/6883050/102499005-f5dcad00-407a-11eb-835b-095e09a02c23.jpg)

**Comment by Aviad-Herman** [comment]
What widget are you using? Are you sure it's from Elementor and not an addon?

**Comment by drosa88** [comment]
> What widget are you using? Are you sure it's from Elementor and not an addon?

The widget in the screenshot is a widget from Ultimate Addons, right. But that makes no difference because the Dynamic Attributes options are set from Elementor core. So the same logic can also be found on rows, columns or native elementor widgets like Text or Headline.

**Comment by Aviad-Herman** [comment]
Ok I see...
The end result you are trying to achieve is a dynamic post listing you can design more freely than the post widget.
Dynamic tags are intended to fetch data from the current post / page and we don't intend to make them into queries.

You will get the design abilities you want once we develop the loop builder feature but it's not going to happen soon.
I hope you will find a hack that works for you in the meanwhile. Update me if you find a good hack!

Keep the good ideas coming :)

**Comment by drosa88** [comment]
Thanks. I will search for a solution and post it there then :-) Have a nice day!

**Comment by drosa88** [comment]
Well. I found a workaround to achieve what I need. It's not perfect, but a good and clean way to handle this.

I´ve created an individual shortcode `project_news` with some parameters:

```
function create_ubznews_shortcode($atts) {
    
    $atts = shortcode_atts(
        array(
        'category' => 'default',
        'offset' => '0',
        'type' => 'title'
        ),
        $atts,
        'ubz_news'
    );
    
    $category = $atts['category'];
    $offset = $atts['offset'];
    $type = $atts['type'];
    $result = "";


    // WP_Query args
    $q_ubz_news_query_args = array(
        'order' => 'DESC',
        'orderby' => 'date',
        'category_name' => $category,
        'offset' => $offset ? $offset : 0,
        'posts_per_page' => 1,
      );
  
      // The Query
      $ubz_news_query = new WP_Query($q_ubz_news_query_args);
      
      // The Loop
      if ( $ubz_news_query->have_posts() ) {
        while ( $ubz_news_query->have_posts() ) {
          $ubz_news_query->the_post();
          
          // Build Result
          if ($type == "title") {
              $result = the_title();
          }
          if ($type == "excerpt") {
              $result = the_excerpt();
          }
          if (type == "featured_image") {
              $result = wp_get_attachment_url( get_post_thumbnail_id($post->ID));
          }
        }
      } else {
        // There are no posts
      }
  
      // Reset Original Post Data
      wp_reset_postdata();
      

    return $result;
    
}
add_shortcode( 'ubz_news', 'create_ubznews_shortcode' );
```

So in Elementor we can use the Dynamic Attribute Shortcode to enter the created one with some params. For example:

![Bildschirmfoto 2020-12-18 um 09 35 26](https://user-images.githubusercontent.com/6883050/102592930-7dc2c580-4114-11eb-83e3-f284f68fe3f5.jpg)

This will give us the correct data from the right loop.

But there is one limitation: The Elementor Dynamic Attribute for background images, for example in columns, does not accept Shortcodes. Here I have to go another way. We´ll see..

**Comment by Aviad-Herman** [comment]
Wow, very cool!
I will refer people with similar questions to your code!

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce the new Loop Builder released with Elementor Pro v3.8.0! 🥳 

With the new Loop Grid widget, you can customize your post listings with custom templates! 💪 
Drag & Drop widgets to create complex designs, fine-tune your Loop Grid layout, add alternate templates, and much more!

This is just the first phase. More customization is coming!

Check out the article here: https://elementor.com/help/loop-grid/

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
